container-storage-setup.1

```diff
@@ -86,10 +86,15 @@ functionality to work properly.
 \f[B]Supported options for the configuration file\f[]:
 
 STORAGE_DRIVER:
-      Specify a storage driver to be used with container runtime.
-      Default: "devicemapper".
-      Valid values are overlay, overlay2 and "".
-      "" tells container-storage-setup to not perform any storage setup.
+      Specify a storage driver to be used with container runtime; the default
+      value will chosen by your OS vendor.
+      Known values are: devicemapper, overlay2, overlay, auto, and "".  The empty string ""
+      is used when you want to ensure container-storage-setup does not automatically
+      change configuration (for example, you want to do it manually).  The "auto" option
+      may be used when c-s-s is started by default (for example, Atomic Host);
+      when specified, c-s-s will choose a backend.  Currently, this will be "overlay2" if
+      supported by the kernel/filesystem at the target mount point, and if not, then
```

Comment:
What is `target mount point`. I understand it. But normal users don't. How about this. We define another knob say `CONTAINER_METADATA_PATH`, which specifies the directory where container metadata will be stored. And STORAGE_DRIVER=auto will make ftype checks on filesystem which has `CONTAINER_METADATA_PATH`.

For compatibility mode, CONTAINER_METADATA_PATH can default to /var/lib/docker. For direct mode, user will have to specify it otherwise we fail. So for case of crio, user will specify `/var/lib/containers/storage` or something like that.

Reply:
It also has the benefit that STORAGE_DRIVER=auto will be properly defined both for compatibility mode and direct mode.